### PR TITLE
docs: update partial hydration doc part

### PR DESCRIPTION
**Pull request by schabibi1**
- [x] Update Partial Hydration (RSC) doc part

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -355,6 +305,37 @@ renderRichText(blok.richTextField, {
 
 ### Gatsby feature references
 
+### Prtial Hydration (RSC)
```
Comment (Dawntraoz) [inline README.md:308]: Small typo here 👆

Comment (schabibi1) [reply README.md:308]: Fixed! Thanks for spotting it @Dawntraoz 🙏 I'll commit it with the rest if there's more coming up from your review :)

File: README.md [hunk 2]
```diff
@@ -54,25 +54,34 @@ Register the plugin on your application and add the [access token](https://www.s
 
 > You need to declare the plugin use and its options in `gatsby-config.js`
 
+`gatsby-config.ts`
```
Comment (Dawntraoz) [inline README.md:57]: Is there a reason to have `gatsby-config.ts` twice? Or maybe it wasn't on purpose? 🤔

File: README.md [hunk 3]
```diff
@@ -120,30 +130,29 @@ storyblokInit({
 
 `gatsby-source-storyblok` does three actions when you initialize it:
 
-- Provides a `useStoryblokState` custom hook in your app, that parses story content JSON into the object.
+- Provides a `StoryblokStory` that includes `useStoryblokState` that receives a story object.
 - Loads [Storyblok Bridge](https://www.storyblok.com/docs/Guides/storyblok-latest-js?utm_source=github.com&utm_medium=readme&utm_campaign=gatsby-source-storyblok) for real-time visual updates.
 - Provides a `storyblokEditable` function to link editable components to the Storyblok Visual Editor.
 
-#### 1. Fetching Content
+> [gatsby-source-storyblok v5](https://github.com/storyblok/gatsby-source-storyblok/tree/v5) and lower had `useStoryblokState` that parsed story content JSON into the object. Use `StoryblokStory` instead to handle the Visual Editor live events when editing story.
```
Comment (Dawntraoz) [inline README.md:137]: Shall this text be about only lower than v5 versions? Or v5 is also using the old `useStoryblokState`?

File: README.md [hunk 4]
```diff
@@ -120,30 +130,29 @@ storyblokInit({
 
 `gatsby-source-storyblok` does three actions when you initialize it:
 
-- Provides a `useStoryblokState` custom hook in your app, that parses story content JSON into the object.
+- Provides a `StoryblokStory` that includes `useStoryblokState` that receives a story object.
```
Comment (Dawntraoz) [inline README.md:133]: As I guess that `StoryblokStory` means using a component here, probably is it better to specify that or at least add `<StoryblokStory />` instead?

File: README.md (hunk 2 above)
Comment (schabibi1) [reply README.md:57]: Yes @Dawntraoz The one you pointed out shows Gatsby plugin configuration for our Gatsby SDK💡
This part below focuses on for Partial Hydration config:

https://github.com/storyblok/gatsby-source-storyblok/pull/340/commits/9c5f70730b6fc4579e41adcb6aff86bbab11d4aa#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R311-R318

File: README.md (hunk 3 above)
Comment (schabibi1) [reply README.md:137]: @Dawntraoz Gatsby SDK v5 uses `useStoryblokState` 👍

File: README.md (hunk 4 above)
Comment (schabibi1) [reply README.md]: Fixed👍 Thank you @Dawntraoz !

File: README.md [hunk 5]
```diff
@@ -120,30 +130,29 @@ storyblokInit({
 
 `gatsby-source-storyblok` does three actions when you initialize it:
 
-- Provides a `useStoryblokState` custom hook in your app, that parses story content JSON into the object.
+- Provides a `<StoryblokStory />` that includes `useStoryblokState` that receives a story object.
```
Comment (alexjoverm) [inline README.md:133]: `useStoryblokState` is internal and user don't even need to know about it. I'd re-phrase it in this lines: `Provides a <StoryblokStory />` component to enable live-editting experience for a story, automatically.

Comment (schabibi1) [reply README.md]: Thanks @alexjoverm ! Fixed it👍

**Pull request merged**